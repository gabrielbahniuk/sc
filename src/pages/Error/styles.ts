import styled from 'styled-components'

export const Container = styled.div`
  margin: 3rem auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  max-width: 1120px;
  width: 80%;

  h1 {
    text-align: center;
    color: var(--text-color);
    font-size: 2rem;
  }
  button {
    width: 80%;
  }
`
