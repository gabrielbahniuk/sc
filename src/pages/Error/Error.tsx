import React from 'react'
import { useInvestments } from '@/hooks'
import { Button } from '@/components/Button'

import { Container } from './styles'

const Error: React.FC = () => {
  const { handleReset } = useInvestments()
  return (
    <Container>
      <h1 data-cy="titleError">Uuups, there was probably a network problem :(</h1>
      <Button type="button" data-cy="btnReset" onClick={handleReset}>
        Let&apos;s try again
      </Button>
    </Container>
  )
}

export { Error }
