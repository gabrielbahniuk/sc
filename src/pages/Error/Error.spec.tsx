import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import { create } from 'react-test-renderer'
import { Error } from './Error'

let wrapper: ShallowWrapper

beforeAll(() => {
  wrapper = shallow(<Error />)
})

afterAll(() => {
  wrapper.unmount()
})

describe('Error Component', () => {
  it('should render Error component', () => {
    const errorComponent = create(<Error />).toJSON()
    expect(errorComponent).toMatchSnapshot()
  })

  it('should render title element', () => {
    expect(wrapper.find('h1').length).toBe(1)
  })
  it('should render button', () => {
    expect(wrapper.find('Button').length).toBe(1)
  })
})
