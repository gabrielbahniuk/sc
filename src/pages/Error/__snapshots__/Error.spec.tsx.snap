// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Component should render Error component 1`] = `
<div
  className="sc-gtsrHT gqsTP"
>
  <h1
    data-cy="titleError"
  >
    Uuups, there was probably a network problem :(
  </h1>
  <button
    className="sc-bdnxRM kjnohf"
    data-cy="btnReset"
  >
    Let's try again
  </button>
</div>
`
